src/ILLink.Shared/DataFlow/ValueSet.cs

```diff
@@ -93,6 +93,8 @@ public void Reset ()
 
 		ValueSet (EnumerableValues values) => _values = values;
 
+		public static implicit operator ValueSet<TValue> (TValue value) => new (value);
```

Comment:
How bad is it if we don't have the implicit conversion? Generally I'd prefer to be explicit, but if this is used in a lot of places it makes sense.

Reply:
It was at least 20 places before I gave up... so probably another 20 after that point :-)

I thought about it and it's actually not surprising. It's a discrepancy between production and consumption.

On the production side a `MultiValue` is always created with just a single value in it - intrinsics, all the visitors/scanners, ... this has to be like that since it needs to create a specific value type (Field access, method parameter, ...). (The only exception is arrays, but those are weird). The only way (except array) we create true multiple-value `MultiValue` is through `Meet`.

On the consumption side everything needs to be able to handle multiple values, so it has to take input as `MultiValue`. Lot of the common helpers are like that. But lot of times these helpers are also called directly from the creation path, which only has single-value at hand. All these places require the conversion. Technically this could be solved by having two overloads for each helper... but that's just messy.

Reply:
Out of curiosity, if I comment out the implicit conversion I get 57 build errors :wink: